Parameters.java

```diff
@@ -263,27 +266,50 @@ public static void load() {
 			if (map.get("sdStep") != null) {
 				sdStep = (double) map.get("sdStep");
 			}
+			if (map.get("meanStepEpitope") != null) {
+				meanStepEpitope = (double) map.get("meanStepEpitope");
+			}
+			if (map.get("sdStepEpitope") != null) {
+				sdStepEpitope = (double) map.get("sdStepEpitope");
+			}
+			if (map.get("epitopeSites") != null) {
+				epitopeSites = toIntArray((List<Integer>) map.get("epitopeSites"));
+			}
+			if (map.get("transitionTransversionRatio") != null) {
+				transitionTransversionRatio = (double) map.get("transitionTransversionRatio");
+			}
 			if (map.get("mut2D") != null) {
 				mut2D = (boolean) map.get("mut2D");
 			}
 			if (map.get("fixedStep") != null) {
 				fixedStep = (boolean) map.get("fixedStep");
 			}
 			if (map.get("startingSequence") != null) {
-				startingSequence = (String) map.get("startingSequence");
-			}
-			if (map.get("alphabetType") != null) {
-				alphabetType = (String) map.get("alphabetType");
+				startingSequence = ((String) map.get("startingSequence")).toUpperCase();
 
-				if (alphabetType.equals("aminoAcids")) {
-					alphabet = AlphabetType.AMINO_ACIDS.getValidCharacters();
+				if (phenotypeSpace.equals("geometricSeq")) {
+					if (startingSequence.length() % 3 != 0) {
+						System.out.println("startingSequence length should be a multiple of 3");
+						throw new IOException();
+					}
+
+					for (int i = 0; i < startingSequence.length(); i += 3) {
+						String triplet = startingSequence.substring(i, i + 3);
+						String translatedAminoAcid = Biology.CodonMap.CODONS.getAminoAcid(triplet);
+
+						if (translatedAminoAcid.equals("STOP")) {
+							System.out.println("There should not be a stop codon at site " + (i/3));
+							throw new IOException();
+						}
+					}
 				}
 			}
-
+			if (map.get("DMSData") != null) {
+				DMSData = (String) map.get("DMSData");
```

Comment:
Could we make sure that the DMS data is the same length (in sites) as the starting sequence (divided by 3 if it's the nucleotide sequence)?

Reply:
This gets tested in `TestGeometricSeqPhenotype.java`, but you make a good point. It would make more sense here. I will remove it from `testDMSData()`!